presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -588,10 +588,18 @@ private PlanWithProperties planTableScan(TableScanNode node, Expression predicat
                     deterministicPredicate,
                     DomainTranslator.toPredicate(newDomain.simplify().transform(assignments::get)));
 
+            LayoutConstraintEvaluator evaluator = new LayoutConstraintEvaluator(
+                    session,
+                    symbolAllocator.getTypes(),
+                    node.getAssignments(),
+                    extractConjuncts(constraint).stream()
+                            .filter(expression -> SymbolsExtractor.extractUnique(expression).stream().noneMatch(context.getCorrelations()::contains))
```

Comment:
s/expression/conjuct

Reply:
Maybe we could extract a method to increase the readability? Like `containsSymbols(Expression, List<Symbols>)`?

Or maybe we could extract whole block as `extractConjunctsNotUsingSymbols(constraint, correlations)`?

Reply:
> s/expression/conjuct

It's still an expression... :)